python_modules/dagster/dagster/cli/dynamic_loader.py

```diff
@@ -208,25 +245,62 @@ def create_pipeline_loading_mode_data(info):
         raise InvalidPipelineLoadingComboError()
 
 
-DynamicObject = namedtuple('DynamicObject', 'module fn module_name fn_name object')
+class DynamicObject:
+    def __init__(self, module, module_name, fn_name):
+        self.module = module
+        self.module_name = module_name
+        self.fn_name = fn_name
+        self.object = None
+        self.coerce_to_repo = False
+        self.loaded = False
+
+    def load(self):
+        if self.loaded:
+            reloader.reload(self.module)
+        self.loaded = True
+
+        fn = getattr(self.module, self.fn_name)
+        check.is_callable(fn)
+        obj = fn()
+
+        if self.coerce_to_repo:
```

Comment:
Can you add an issue noting that this object being aware of Repository and Pipeline is not and then link to that issue from a comment here. Thanks

Reply:
Yep added a comment